syndeo-template/src/main/java/com/google/cloud/syndeo/dlq/pubsub/PubsubDlqSchemaTransformProvider.java

```diff
@@ -0,0 +1,135 @@
+public class PubsubDlqSchemaTransformProvider
+    extends TypedSchemaTransformProvider<PubsubDlqWriteConfiguration> {
+
+  @Override
+  public Class<PubsubDlqWriteConfiguration> configurationClass() {
+    return PubsubDlqWriteConfiguration.class;
+  }
+
+  @Override
+  public SchemaTransform from(PubsubDlqWriteConfiguration configuration) {
+    return new PubsubDlqSchemaTransform(configuration);
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized String identifier() {
+    return "syndeo:schematransform:com.google.cloud:pubsub_dlq_write:v1";
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized List<@UnknownKeyFor @NonNull @Initialized String>
+      inputCollectionNames() {
+    return Collections.singletonList("input");
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized List<@UnknownKeyFor @NonNull @Initialized String>
+      outputCollectionNames() {
+    return Collections.emptyList();
+  }
+  /**
+   * An implementation of {@link SchemaTransform} for Pub/Sub writes configured using {@link
+   * PubsubDlqWriteConfiguration}.
+   */
+  static class PubsubDlqSchemaTransform extends PTransform<PCollectionRowTuple, PCollectionRowTuple>
+      implements SchemaTransform {
+
+    public static final int PUBSUB_MESSAGE_SIZE = 10 * 1024 * 1024;
+    private final PubsubDlqWriteConfiguration configuration;
+
+    private PubsubClient.PubsubClientFactory pubsubClientFactory;
+
+    PubsubDlqSchemaTransform(PubsubDlqWriteConfiguration configuration) {
+      this.configuration = configuration;
+    }
+
+    PubsubDlqSchemaTransform withPubsubClientFactory(PubsubClient.PubsubClientFactory factory) {
+      this.pubsubClientFactory = factory;
+      return this;
+    }
+
+    /** Implements {@link SchemaTransform} buildTransform method. */
+    @Override
+    public PTransform<PCollectionRowTuple, PCollectionRowTuple> buildTransform() {
+      return this;
+    }
+
+    @Override
+    public PCollectionRowTuple expand(PCollectionRowTuple input) {
+      PCollection<PubsubMessage> messages =
+          input.get("errors").apply(ParDo.of(new ConvertRowToMessage()));
+
+      PubsubIO.Write<PubsubMessage> write = PubsubIO.writeMessages().to(configuration.getTopic());
+      if (pubsubClientFactory != null) {
+        write = write.withClientFactory(pubsubClientFactory);
+      }
+      messages.apply(PubsubIO.Write.class.getSimpleName(), write);
+
+      return PCollectionRowTuple.empty(input.getPipeline());
+    }
+
+    public static class ConvertRowToMessage extends DoFn<Row, PubsubMessage> {
+      @ProcessElement
+      public void process(@Element Row row, OutputReceiver<PubsubMessage> receiver) {
+        byte[] message = row.toString(true).getBytes(StandardCharsets.UTF_8);
+        if (message.length < PUBSUB_MESSAGE_SIZE) {
+          receiver.output(new PubsubMessage(message, Map.of()));
+        } else {
+          byte[] truncated = new byte[PUBSUB_MESSAGE_SIZE];
```

Comment:
have you tested that the proper cutoff is the exact 10mb?

Reply:
yes, this is exactly 10mb. have added a test that kind of checks this too.